apps/google-analytics-4/frontend/src/components/main-app/analytics-app/AnalyticsApp.tsx

```diff
@@ -1,38 +1,54 @@
 import { useAutoResizer, useFieldValue } from '@contentful/react-apps-toolkit';
 import ChartFooter from 'components/main-app/ChartFooter';
 import ChartHeader from 'components/main-app/ChartHeader';
-import { useEffect, useState } from 'react';
-import { DateRangeType, Row, RunReportResponse } from 'types';
-import DateRange from 'helpers/dateRange.enum';
-import { config } from 'config';
+import { useEffect, useState, useMemo } from 'react';
+import { useApi } from 'hooks/useApi';
+import getRangeDates from 'helpers/handle-date-range/handle-date-range';
+import ChartContent from '../ChartContent';
+import { DateRangeType, ServiceAccountKeyId, ServiceAccountKey } from 'types';
 import { styles } from './AnalyticsApp.styles';
-import ChartContent from 'components/main-app/ChartContent';
 import { Flex, Note } from '@contentful/f36-components';
+import { RunReportData } from '@/apis/apiTypes';
 
 const DEFAULT_ERR_MSG = 'Oops! Cannot display the analytics data at this time.';
 const EMPTY_DATA_MSG = 'There are no page views to show for this range';
 
-const AnalyticsApp = () => {
-  const [runReportResponse, setRunReportResponse] = useState<RunReportResponse>(
-    {} as RunReportResponse
-  );
-  const [pageViewData, setPageViewData] = useState<RunReportResponse>(runReportResponse);
+interface Props {
+  serviceAccountKeyId: ServiceAccountKeyId;
+  serviceAccountKey: ServiceAccountKey;
+  propertyId: string;
+  reportSlug: string;
+}
+const AnalyticsApp = (props: Props) => {
+  const { serviceAccountKeyId, serviceAccountKey, propertyId, reportSlug } = props;
+  const [runReportResponse, setRunReportResponse] = useState<RunReportData>({} as RunReportData);
   const [dateRange, setDateRange] = useState<DateRangeType>('lastWeek');
+  const [startEndDates, setStartEndDates] = useState<any>(getRangeDates('lastWeek')); // TYPE
```

Comment:
Can we create an Object that maps the string like 
```
const DATE_RANGES = {
  lastWeek: 'lastWeek',
  lastMonth: 'lastMonth',
}
```
Then we can just call those values instead of writing those strings in every time

Reply:
Found it, it's in `handle-date-range.ts`

Reply:
will rename that file to be more generalized ^^^ will make it easier to find!